Fix #17068 Progressbars must have accessible name in Contributor dashboard

PVenkatArvind opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #17068.
2. This PR does the following: [fixes Progressbars must have accessible name in Contributor dashboard]

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct
before
![conntrib_17068](https://user-images.githubusercontent.com/79320301/223705683-ad75a506-f5a1-4e16-9000-89f6451a5d0f.png)


after
![17068_after](https://user-images.githubusercontent.com/79320301/223705517-09dc61e5-99a0-4710-8815-2c82f3b6b2fa.png)

https://user-images.githubusercontent.com/79320301/223705729-3ee43ac9-9d03-4507-adeb-5682554938a3.MP4


<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly. If this PR is for a developer-facing feature,
provide the videos/screenshots of developer-facing interface.

Please also include videos/screenshots of the developer tools
browser console, so that we can be sure that there are no console errors.

If the changes in your PRs are autogenerated via a script and you cannot provide proof
for the changes then please leave a comment "No proof of changes needed because {{Reason}}"
and remove all the sections below.
-->

#### Proof of changes on desktop with slow/throttled network

<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->

#### Proof of changes on mobile phone

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

PVenkatArvind committed:
fix_progressbar_chromevox

sagangwee commented on core/templates/pages/contributor-dashboard-page/opportunities-list-item/opportunities-list-item.component.html:
```diff
@@ -7,6 +7,15 @@
            class="progress">
         <div class="progress-bar bg-success"
              role="progressbar"
+             [style.width.%]="progressValue"
+             [attr.aria-valuenow]="progressValue !== undefined ? progressValue.toString() : ''"
+             [attr.aria-valuemin]="0"
+             [attr.aria-valuemax]="100"
+             [attr.aria-label]="progressBarLabel"
+             title="progressBar"
+             ngbTooltip="{{opportunity?.translationsCount}}/{{opportunity?.totalCount}} cards accepted"
+             container="body"
+             placement="bottom"
```
Don't think most of these attributes are necessary, just a `aria-label` tag here and in the 3 types of progress bars below.

PVenkatArvind replied:
Noted!

PVenkatArvind replied:
will resolve and request for review

sagangwee reviewed (changes requested):
@PVenkatArvind Thanks, just one comment.

PVenkatArvind committed:
Merge branch 'develop' into progressbar_17068

PVenkatArvind committed:
fix_progressbar

PVenkatArvind committed:
Merge branch 'develop' into progressbar_17068

PVenkatArvind committed:
Merge branch 'develop' into progressbar_17068

PVenkatArvind committed:
Merge branch 'develop' into progressbar_17068

PVenkatArvind committed:
Merge branch 'oppia:develop' into progressbar_17068

PVenkatArvind committed:
Merge branch 'oppia:develop' into progressbar_17068

chris7716 commented:
@sagangwee PTAL thanks!

sagangwee commented on core/templates/pages/contributor-dashboard-page/opportunities-list-item/opportunities-list-item.component.html:
```diff
@@ -7,6 +7,11 @@
            class="progress">
         <div class="progress-bar bg-success"
              role="progressbar"
+             [attr.aria-label]="progressBarLabel"
```
Why `[attr.aria-label]` instead of just `aria-label`?

PVenkatArvind replied:
https://angular.io/guide/attribute-binding   --->Angular documentation thats reason i have used it

attr.aria-label used because in Angular to bind an aria-label attribute dynamically to an element using attribute binding and  Angular will update the value of the attribute whenever the value of the property changes if we use only 
aria-label we should manually  set the label based on some value in the component,

sagangwee replied:
Gotcha, thanks!

sagangwee commented on core/templates/pages/contributor-dashboard-page/opportunities-list-item/opportunities-list-item.component.html:
```diff
@@ -7,6 +7,11 @@
         <div class="progress-bar bg-success"
              role="progressbar"
+             [attr.aria-label]="progressBarLabel"
+             title="progressBar"
+             ngbTooltip="{{opportunity?.translationsCount}}/{{opportunity?.totalCount}} cards accepted"
+             container="body"
+             placement="bottom"
```
Are these even necessary here? Also, note that the tooltip description is incorrect since this is the progress bar for practice questions not translations.

PVenkatArvind replied:
Yes , Sorry for the inconvenience. The reason i added the ngbtooltip was due to *ngIf is a structural directive *ngIf="!translationProgressBar" due to which i have pasted the same tooltip below  that was used in the codebase  i should have made it ngbTooltip="{{opportunity?.progressPercentage}}% of practice questions completed" something anyway 
i will remove this line and make a commit Thanks a lot  @sagangwee

sagangwee reviewed (changes requested):
Thanks, took a pass.

PVenkatArvind committed:
Merge branch 'develop' into progressbar_17068

PVenkatArvind committed:
fix_progressbar

Ash-2k3 commented:
@sagangwee PTAL, Thanks

PVenkatArvind committed:
Merge branch 'develop' into progressbar_17068

sagangwee commented on core/templates/pages/contributor-dashboard-page/opportunities-list-item/opportunities-list-item.component.html:
```diff
@@ -7,6 +7,10 @@
         <div class="progress-bar bg-success"
              role="progressbar"
+             [attr.aria-label]="progressBarLabel"
+             title="progressBar"
+             container="body"
+             placement="bottom"
```
Sorry, but are we sure these attributes are necessary?

PVenkatArvind replied:
Greetings ! @sagangwee 
Yes it was added because Chromevox reads the title name .
Hence  I added the same for accessibility.

sagangwee replied:
Okay, but what about `container` and `placement`?

sagangwee reviewed (changes requested):
@PVenkatArvind Thanks, just one comment.

Ash-2k3 commented:
Hey @PVenkatArvind, have you addressed @sagangwee 's comments ? If yes you may assign him to review the changes. Thanks

PVenkatArvind commented:
> Hey @PVenkatArvind, have you addressed @sagangwee 's comments ? If yes you may assign him to review the changes. Thanks
Greetings! @Ash-2k3 I have addressed the comments and requested the same 
Thanks !

Ash-2k3 commented:
> > Hey @PVenkatArvind, have you addressed @sagangwee 's comments ? If yes you may assign him to review the changes. Thanks
> > Greetings! @Ash-2k3 I have addressed the comments and requested the same
> > Thanks !

As a future note, to assign a reviewer, you can comment '@xxx PTAL', and the Oppia bot will automatically assign the person mentioned.

PVenkatArvind commented:
> > > Hey @PVenkatArvind, have you addressed @sagangwee 's comments ? If yes you may assign him to review the changes. Thanks
> > > Greetings! @Ash-2k3 I have addressed the comments and requested the same
> > > Thanks !
> 
> As a future note, to assign a reviewer, you can comment '@xxx PTAL', and the Oppia bot will automatically assign the person mentioned.

Sorry for the inconvenience caused , although I have manually re requested the reviewer for the same 
Thanks @Ash-2k3

PVenkatArvind committed:
Merge branch 'develop' into progressbar_17068

PVenkatArvind committed:
Merge branch 'develop' into progressbar_17068

sagangwee reviewed (changes requested):
@PVenkatArvind

PVenkatArvind commented:
> @PVenkatArvind

@sagangwee  I have added the same i.e. container and placement for the  tooltip, I forgot to remove the same when you pointed out the necessity of the tooltip .
Thanks !

PVenkatArvind committed:
fix_progressbar

PVenkatArvind committed:
Merge branch 'oppia:develop' into progressbar_17068

PVenkatArvind committed:
Merge branch 'develop' into progressbar_17068

sagangwee reviewed (approved):
@PVenkatArvind LGTM, thanks!

PVenkatArvind committed:
Merge branch 'develop' into progressbar_17068

Ash-2k3 commented:
@seanlip, I am not sure what happened here. Can you PTAL ?

seanlip commented:
Thanks for flagging, @Ash-2k3. There seems to have been some issue in the "checkout" action, see https://github.com/oppia/oppia/actions/runs/4681724982/jobs/8295952655. /cc @U8NWXD @gp201 FYI.

I'll try merging it again.

seanlip commented:
OK, I am not sure what is going on here (with the merge queue). @U8NWXD @gp201 could you please advise?

gp201 commented:
@seanlip It could either beacuse this action failed https://github.com/oppia/oppia/actions/runs/4687437529/jobs/8307612081

Or because the actions don't have any runners free to take up the job. Merge queue removes PRs if the actions are not completed within 4 hours.

seanlip commented:
Oh, I missed the failure, thanks. OK, let me try merging again, but you might want to look into this if it fails again...
